doc/api/net.md

```diff
@@ -416,109 +571,134 @@ See also: [`socket.setTimeout()`][]
+currently pending in the `net.Socket` instances write buffer.
+
+The `net.Socket` class has been designed such that calls to `socket.write()`
+will always work so long as the connection is open. Data written to the socket
+is buffered internally first, then written out to the underlying connection. If
+the data is written to the buffer faster than it can be passed to the connection
+(e.g. when using a slow network connection), the size of the internal buffer
+can grow.
 
-The consequence of this internal buffering is that memory may grow. This
-property shows the number of characters currently buffered to be written.
-(Number of characters is approximately equal to the number of bytes to be
-written, but the buffer may contain strings, and the strings are lazily
-encoded, so the exact number of bytes is not known.)
+The value of `socket.bufferSize` represents the amount of data (in bytes)
+pending in the queue.
 
-Users who experience large or growing `bufferSize` should attempt to
-"throttle" the data flows in their program with [`pause()`][] and [`resume()`][].
+In order to avoid large or growing `socket.bufferSize` numbers, attempts should
+be made to throttle the flow of data written to the socket.
 
 ### socket.bytesRead
 <!-- YAML
 added: v0.5.3
 -->
 
-The amount of received bytes.
+The `socket.bytesRead` property specifies the total number of bytes received.
 
 ### socket.bytesWritten
 <!-- YAML
 added: v0.5.3
 -->
 
-The amount of bytes sent.
+The `socket.bytesWritten` property specifies the total number of bytes written.
 
 ### socket.connect(options[, connectListener])
 <!-- YAML
 added: v0.1.90
 -->
 
-Opens the connection for a given socket.
-
-For TCP sockets, `options` argument should be an object which specifies:
-
-  - `port`: Port the client should connect to (Required).
-
-  - `host`: Host the client should connect to. Defaults to `'localhost'`.
+* `options` {Object}
+  * `path`: Path the client should connect to
+  * `port` {number} Port the client should connect to
+  * `host` {String} Host the client should connect to
+     Defaults to `'localhost'`
+  * `localAddress` {String} Local interface to bind to for network connections
+  * `localPort` {number} Local port to bind to for network connections
+  * `family` {number} Version of IP stack. Defaults to `4`
+  * `hints` {number} [`dns.lookup()` hints][]. Defaults to `0`
+  * `lookup` {Function} Custom lookup function. Defaults to `dns.lookup`
+* `connectListener` {Function}
 
-  - `localAddress`: Local interface to bind to for network connections.
+The `socket.connect()` method opens the connection for a given socket using
+either the given `path`, or the `port` and `host`.
 
-  - `localPort`: Local port to bind to for network connections.
+In most normal situations it is not necessary to call this method directly
+as other methods such as `net.createConnection()` will call `socket.connect()`
+automatically.
```

Comment:
its possible to call it multiple times, to reconnect (pretty sure), is that described here?